docs(DWH): feature branch for DWH (#4766)
scaleway/docs-content: pages/data-warehouse/quickstart.mdx

---
meta:
  title: Data Warehouse for ClickHouse® - Quickstart
  description: This page shows you how to get started with Scaleway's Data Warehouse for ClickHouse®.
content:
  h1: Data Warehouse for ClickHouse® - Quickstart
  paragraph: This page shows you how to get started with Scaleway's Data Warehouse for ClickHouse®.
tags: data warehouse quickstart clickhouse example guide step guide
dates:
  validation: 2025-05-19
  posted: 2025-05-19
categories:
  - data-warehouse
---

Data Warehouse for ClickHouse® is a centralized repository to store large sets of structured data, to streamline analysis and reporting for data science and business intelligence.

<Macro id="requirements" />

- A Scaleway account logged into the [console](https://console.scaleway.com)
- [Owner](/iam/concepts/#owner) status or [IAM permissions](/iam/concepts/#permission) allowing you to perform actions in the intended Organization
- [Signed up to the private beta](https://www.scaleway.com/fr/betas/) and received a confirmation email.

## How to create a Data Warehouse for ClickHouse® deployment

> 1. Click **ClickHouse®** under **Managed Services** on the side menu. The Managed Data Warehouse for ClickHouse® page displays.

Reviewer: ```suggestion
1. Click **ClickHouse®** under **Data & Analytics** on the side menu. The Managed Data Warehouse for ClickHouse® page displays.
```

3. Click **+ Create deployment**. The Data Warehouse deployment creation wizard displays.

> 4. Select the **region** in which to create your deployment.

Reviewer: ```suggestion
3. Select the **region** in which to create your deployment.
```

5. Select the **For basic developments** preset.
    <Message type="note">
    You can update the autoscaling configuration of your deployment at any time after its creation.
    </Message>

> 6. Enter a name for your deployment, or keep the automatically generated one.

Reviewer: ```suggestion
5. Enter a name for your deployment, or keep the automatically generated one.
```

7. Configure the administrator credentials for your deployment by entering a password, or by generating a random one.
    <Message type="note">
    The username cannot be modified.
    </Message>

> 9. Review the estimated cost, then click **Create deployment** to confirm. A list of your Data Warehouse for ClickHouse® deployments displays, showing the newly created resources.

Reviewer: ```suggestion
7. Review the estimated cost, then click **Create deployment** to confirm. A list of your Data Warehouse for ClickHouse® deployments displays, showing the newly created resources.
```

## How to connect to your deployment

### Connecting using the ClickHouse® HTTP console

1. From the Data Warehouse for ClickHouse® deployments page, click the name of the deployment you want to want to connect to. The **Overview** tab displays.

2. Click the **ClickHouse® HTTP console** button. A new tab containing the console displays.

> 3. In the top-corner of the ClickHouse® HTTP console, replace the `default` login by `scwadmin`.

Reviewer: ```suggestion
3. In the top corner of the ClickHouse® HTTP console, replace the `default` login by `scwadmin`.
```

    INSERT INTO nyc_taxi.trips_small
    SELECT
        trip_id,
        pickup_datetime,
        dropoff_datetime,
        pickup_longitude,
        pickup_latitude,
        dropoff_longitude,
        dropoff_latitude,
        passenger_count,
        trip_distance,
        fare_amount,
        extra,
        tip_amount,
        tolls_amount,
        total_amount,
        payment_type,
        pickup_ntaname,
        dropoff_ntaname
    FROM s3(
        'https://datasets-documentation.s3.eu-west-3.amazonaws.com/nyc-taxi/trips_{0..2}.gz',
        'TabSeparatedWithNames'
    );
    ```

### Querying the database

1. Run the command below to count the rows you inserted in the table:
    ```sql
    SELECT count()
    FROM nyc_taxi.trips_small;
    ```

2. Run the command below to list the 10 first rows of your table:
    ```sql
    SELECT *
    FROM nyc_taxi.trips_small
    LIMIT 10;
    ```

3. Run the command below to display the 10 first neighborhoods with the most frequent pickups:
    ```sql
    SELECT
      pickup_ntaname,
      count(*) AS count
    FROM nyc_taxi.trips_small WHERE pickup_ntaname != ''
    GROUP BY pickup_ntaname
    ORDER BY count DESC
    LIMIT 10;
    ```

> To perform more in-depth tests with larger data sets, refer to our [dedicated documentation](/data-warehouse/reference-content/example-datasets/)

Reviewer: ```suggestion
To perform more in-depth tests with larger data sets, refer to our [dedicated documentation](/data-warehouse/reference-content/example-datasets/).
```

## How to manage your deployment

1. From the Data Warehouse for ClickHouse® deployments page, click the name of the deployment you want to manage. The overview tab of the deployment displays. From this view, you can see the configuration of your deployment.

2. Click the **Settings** tab. From this view, you can:

>   - Access the CLickHouse® HTTP console

Reviewer: ```suggestion
  - Access the ClickHouse® HTTP console
```

  - Connect to your deployment using frameworks
  - Connect your deployment to your preferred BI tools
  - [Delete your Data Warehouse deployment](/data-warehouse/how-to/manage-delete-deployment/#how-to-delete-a-data-warehouse).

3. Click the **Databases** tab. From this view you can:

    - View the databases you created in your deployment.

4. Click the **Users** tab. From this view, you can:

>     - view the users in your deployment.

Reviewer: ```suggestion
    - View the users in your deployment.
```

>     - modify the password of a user.

Author: ```suggestion
    - modify the password of a user.
```
```suggestion
    - View the users in your deployment.
    - Modify the password of a user.
```

## How to delete your deployment

<Message type="important">
  This action is irreversible and will permanently delete this Data Warehouse deployment and all its associated data.
</Message>

> 1. Click **Data Warehouse** under **Managed Services** on the side menu. The Data Warehouse deployment page displays.

Reviewer: ```suggestion
1. Click **ClickHouse®** under **Data & Analytics** on the side menu. The Data Warehouse deployment page displays.
```
